### PR TITLE
Add flux example

### PR DESCRIPTION
**Pull request by clackhan**
## Summary by CodeRabbit

- **New Features**
	- Introduced a comprehensive README for the Flux example, detailing environment setup and execution instructions.
	- Added a command-line interface for generating images using the Flux model, allowing users to specify parameters and optimization flags.

- **Documentation**
	- Enhanced README with sections on performance comparison, dynamic shape support, and quality of generated images.
	- Included detailed instructions for setting up OneDiff, the NexFort backend, and the Diffusers library.

### CODE REVIEW

File: onediff_diffusers_extensions/examples/flux/README.md [hunk 1]
```diff
@@ -0,0 +1,110 @@
...
+    --run_multiple_resolutions 1 \
+    --saved-image flux_compile.png
+```
+
+## Quality
+When using nexfort as the backend for onediff compilation acceleration, the generated images are lossless.
+
+<p align="center">
+<img src="../../../imgs/nexfort_flux_demo.png">
+</p>
```
Comment (strint) [inline onediff_diffusers_extensions/examples/flux/README.md:110]: 这里是不是应该用下 A100，做下和 base 的对比，一共两张图

File: onediff_diffusers_extensions/examples/flux/text_to_image_flux.py [hunk 2]
```diff
@@ -0,0 +1,233 @@
...
+                self.pipe.text_encoder_2 = quantize(
+                    self.pipe.text_encoder_2,
+                    quant_type="fp8_e4m3_e4m3_dynamic_per_tensor",
+                )
+
+        self.pipe.to(device)
+
+        if enable_fast_transformer:
+            print("compile...")
+            from nexfort.compilers.transform_model import transform_model
```
Comment (strint) [inline onediff_diffusers_extensions/examples/flux/text_to_image_flux.py:151]: 这个接口会经常用，看是不是导出一个 transform 接口，这样接口比较简单

File: onediff_diffusers_extensions/examples/flux/README.md (hunk 1 above)
Comment (clackhan) [reply onediff_diffusers_extensions/examples/flux/README.md:110]: > 这里是不是应该用下 A100，做下和 base 的对比，一共两张图

我们现在没有A100机器

File: onediff_diffusers_extensions/examples/flux/text_to_image_flux.py (hunk 2 above)
Comment (clackhan) [reply onediff_diffusers_extensions/examples/flux/text_to_image_flux.py:151]: > 这个接口会经常用，看是不是增加一个简化的 `transform` 接口，这样 API 比较简单

可以，nexfort也同步改一下

File: onediff_diffusers_extensions/examples/flux/README.md [hunk 3]
```diff
@@ -0,0 +1,110 @@
...
+## Performance comparation
+
+```
+NEXFORT_FORCE_QUANTE_ON_CUDA=1 python3 onediff_diffusers_extensions/examples/flux/text_to_image_flux.py \
+    --quantize \
+    --fast-transform \
+    --saved-image flux_compile.png
+```
+
+Testing on NVIDIA H20, with image size of 1024*1024, iterating 20 steps:
```
Comment (strint) [inline onediff_diffusers_extensions/examples/flux/README.md:51]: H20 上看起来适合拆成两个结果，transform 一个、transform + 量化一个，前者是社区版，后者是企业版

File: onediff_diffusers_extensions/examples/flux/README.md (hunk 1 above)
Comment (clackhan) [reply onediff_diffusers_extensions/examples/flux/README.md:116]: > 这里是不是应该用下 A100，做下和 base 的对比，一共两张图

已添加和base的对比图

File: onediff_diffusers_extensions/examples/flux/README.md (hunk 3 above)
Comment (clackhan) [reply onediff_diffusers_extensions/examples/flux/README.md:75]: > H20 上看起来适合拆成两个结果，transform 一个、transform + 量化一个，前者是社区版，后者是企业版

已拆

**Review by strint — APPROVED** [review]

**Comment by liho00** [comment]
from nexfort.compilers import transform
0|comfy-0  | ImportError: cannot import name 'transform' from 'nexfort.compilers' (/home/shadeform/miniconda3/envs/image/lib/python3.10/site-packages/nexfort/compilers/init.cpython-310-x86_64-linux-gnu.so)

Name: nexfort
Version: 0.1.dev291
Summary: Next Generation Extension for PyTorch

**Comment by liho00** [comment]
and can we have comfyui example as well?

**Comment by strint** [comment]
> from nexfort.compilers import transform 0|comfy-0 | ImportError: cannot import name 'transform' from 'nexfort.compilers' (/home/shadeform/miniconda3/envs/image/lib/python3.10/site-packages/nexfort/compilers/init.cpython-310-x86_64-linux-gnu.so)
> 
> Name: nexfort Version: 0.1.dev291 Summary: Next Generation Extension for PyTorch
> 
> @strint @clackhan

@liho00 Please update nexfort https://pypi.org/project/nexfort/#history

**Comment by liho00** [comment]
> > from nexfort.compilers import transform 0|comfy-0 | ImportError: cannot import name 'transform' from 'nexfort.compilers' (/home/shadeform/miniconda3/envs/image/lib/python3.10/site-packages/nexfort/compilers/init.cpython-310-x86_64-linux-gnu.so)
> > Name: nexfort Version: 0.1.dev291 Summary: Next Generation Extension for PyTorch
> > @strint @clackhan
> 
> @liho00 Please update nexfort https://pypi.org/project/nexfort/#history

```
    pipe.transformer = transform(pipe.transformer)
  File "src/nexfort/compilers/transform_model.py", line 42, in nexfort.compilers.transform_model.transform_model
  File "src/nexfort/compilers/register_diffusers.py", line 15, in init nexfort.compilers.register_diffusers
  File "src/nexfort/nn/modules.py", line 14, in init nexfort.nn.modules
  File "src/nexfort/kernels/activations.py", line 43, in init nexfort.kernels.activations
  File "/home/shadeform/miniconda3/envs/image/lib/python3.10/site-packages/triton/runtime/jit.py", line 882, in jit
    return decorator(fn)
  File "/home/shadeform/miniconda3/envs/image/lib/python3.10/site-packages/triton/runtime/jit.py", line 871, in decorator
    return JITFunction(
  File "/home/shadeform/miniconda3/envs/image/lib/python3.10/site-packages/triton/runtime/jit.py", line 704, in __init__
    self.starting_line_number = inspect.getsourcelines(fn)[1]
  File "/home/shadeform/miniconda3/envs/image/lib/python3.10/inspect.py", line 1121, in getsourcelines
    lines, lnum = findsource(object)
  File "/home/shadeform/miniconda3/envs/image/lib/python3.10/inspect.py", line 940, in findsource
    file = getsourcefile(object)
  File "/home/shadeform/miniconda3/envs/image/lib/python3.10/inspect.py", line 817, in getsourcefile
    filename = getfile(object)
  File "/home/shadeform/miniconda3/envs/image/lib/python3.10/site-packages/torch/package/package_importer.py", line 712, in _patched_getfile
    return _orig_getfile(object)
  File "/home/shadeform/miniconda3/envs/image/lib/python3.10/inspect.py", line 797, in getfile
    raise TypeError('module, class, method, function, traceback, frame, or '
TypeError: module, class, method, function, traceback, frame, or code object was expected, got cython_function_or_method
```

**Comment by clackhan** [comment]
> > > from nexfort.compilers import transform 0|comfy-0 | ImportError: cannot import name 'transform' from 'nexfort.compilers' (/home/shadeform/miniconda3/envs/image/lib/python3.10/site-packages/nexfort/compilers/init.cpython-310-x86_64-linux-gnu.so)
> > > Name: nexfort Version: 0.1.dev291 Summary: Next Generation Extension for PyTorch
> > > @strint @clackhan
> > 
> > 
> > @liho00 Please update nexfort https://pypi.org/project/nexfort/#history
> 
> @strint @clackhan got new error
> 
> Name: torch Version: 2.4.1
> 
> Name: nexfort Version: 0.1.dev293
> 
> ```
> NEXFORT_ENABLE_FP8_QUANTIZE_ATTENTION=0 python3 onediff_diffusers_extensions/examples/flux/text_to_image_flux.py     --transform     --saved-image flux_compile.png
> /home/shadeform/miniconda3/envs/image/lib/python3.10/site-packages/onnxscript/converter.py:820: FutureWarning: 'onnxscript.values.Op.param_schemas' is deprecated in version 0.1 and will be removed in the future. Please use '.op_signature' instead.
>   param_schemas = callee.param_schemas()
> /home/shadeform/miniconda3/envs/image/lib/python3.10/site-packages/onnxscript/converter.py:820: FutureWarning: 'onnxscript.values.OnnxFunction.param_schemas' is deprecated in version 0.1 and will be removed in the future. Please use '.op_signature' instead.
>   param_schemas = callee.param_schemas()
> Loading checkpoint shards: 100%|███████████████████████████████████████████████| 2/2 [00:00<00:00, 14.00it/s]
> Loading pipeline components...:  43%|██████████████████                        | 3/7 [00:00<00:00, 14.61it/s]You set `add_prefix_space`. The tokenizer needs to be converted from the slow tokenizers
> Loading pipeline components...: 100%|██████████████████████████████████████████| 7/7 [00:00<00:00,  8.65it/s]
> compile...
> Traceback (most recent call last):
>   File "/home/shadeform/onediff/onediff_diffusers_extensions/examples/flux/text_to_image_flux.py", line 231, in <module>
>     main()
>   File "/home/shadeform/onediff/onediff_diffusers_extensions/examples/flux/text_to_image_flux.py", line 186, in main
>     flux = FluxGenerator(args.model, args.quantize, args.transform, args.speedup_t5)
>   File "/home/shadeform/onediff/onediff_diffusers_extensions/examples/flux/text_to_image_flux.py", line 153, in __init__
>     self.pipe.transformer = transform(self.pipe.transformer)
>   File "src/nexfort/compilers/transform_model.py", line 42, in nexfort.compilers.transform_model.transform_model
>   File "src/nexfort/compilers/register_diffusers.py", line 15, in init nexfort.compilers.register_diffusers
>   File "src/nexfort/nn/modules.py", line 14, in init nexfort.nn.modules
>   File "src/nexfort/kernels/activations.py", line 43, in init nexfort.kernels.activations
>   File "/home/shadeform/miniconda3/envs/image/lib/python3.10/site-packages/triton/runtime/jit.py", line 882, in jit
>     return decorator(fn)
>   File "/home/shadeform/miniconda3/envs/image/lib/python3.10/site-packages/triton/runtime/jit.py", line 871, in decorator
>     return JITFunction(
>   File "/home/shadeform/miniconda3/envs/image/lib/python3.10/site-packages/triton/runtime/jit.py", line 704, in __init__
>     self.starting_line_number = inspect.getsourcelines(fn)[1]
>   File "/home/shadeform/miniconda3/envs/image/lib/python3.10/inspect.py", line 1121, in getsourcelines
>     lines, lnum = findsource(object)
>   File "/home/shadeform/miniconda3/envs/image/lib/python3.10/inspect.py", line 940, in findsource
>     file = getsourcefile(object)
>   File "/home/shadeform/miniconda3/envs/image/lib/python3.10/inspect.py", line 817, in getsourcefile
>     filename = getfile(object)
>   File "/home/shadeform/miniconda3/envs/image/lib/python3.10/site-packages/torch/package/package_importer.py", line 712, in _patched_getfile
>     return _orig_getfile(object)
>   File "/home/shadeform/miniconda3/envs/image/lib/python3.10/inspect.py", line 797, in getfile
>     raise TypeError('module, class, method, function, traceback, frame, or '
> TypeError: module, class, method, function, traceback, frame, or code object was expected, got cython_function_or_method
> ```

please install torch==2.4.0 and try Again

**Comment by liho00** [comment]
> > > > from nexfort.compilers import transform 0|comfy-0 | ImportError: cannot import name 'transform' from 'nexfort.compilers' (/home/shadeform/miniconda3/envs/image/lib/python3.10/site-packages/nexfort/compilers/init.cpython-310-x86_64-linux-gnu.so)
> > > > Name: nexfort Version: 0.1.dev291 Summary: Next Generation Extension for PyTorch
> > > > @strint @clackhan
> > > 
> > > 
> > > @liho00 Please update nexfort https://pypi.org/project/nexfort/#history
> > 
> > 
> > @strint @clackhan got new error
> > Name: torch Version: 2.4.1
> > Name: nexfort Version: 0.1.dev293
> > ```
> > NEXFORT_ENABLE_FP8_QUANTIZE_ATTENTION=0 python3 onediff_diffusers_extensions/examples/flux/text_to_image_flux.py     --transform     --saved-image flux_compile.png
> > /home/shadeform/miniconda3/envs/image/lib/python3.10/site-packages/onnxscript/converter.py:820: FutureWarning: 'onnxscript.values.Op.param_schemas' is deprecated in version 0.1 and will be removed in the future. Please use '.op_signature' instead.
> >   param_schemas = callee.param_schemas()
> > /home/shadeform/miniconda3/envs/image/lib/python3.10/site-packages/onnxscript/converter.py:820: FutureWarning: 'onnxscript.values.OnnxFunction.param_schemas' is deprecated in version 0.1 and will be removed in the future. Please use '.op_signature' instead.
> >   param_schemas = callee.param_schemas()
> > Loading checkpoint shards: 100%|███████████████████████████████████████████████| 2/2 [00:00<00:00, 14.00it/s]
> > Loading pipeline components...:  43%|██████████████████                        | 3/7 [00:00<00:00, 14.61it/s]You set `add_prefix_space`. The tokenizer needs to be converted from the slow tokenizers
> > Loading pipeline components...: 100%|██████████████████████████████████████████| 7/7 [00:00<00:00,  8.65it/s]
> > compile...
> > Traceback (most recent call last):
> >   File "/home/shadeform/onediff/onediff_diffusers_extensions/examples/flux/text_to_image_flux.py", line 231, in <module>
> >     main()
> >   File "/home/shadeform/onediff/onediff_diffusers_extensions/examples/flux/text_to_image_flux.py", line 186, in main
> >     flux = FluxGenerator(args.model, args.quantize, args.transform, args.speedup_t5)
> >   File "/home/shadeform/onediff/onediff_diffusers_extensions/examples/flux/text_to_image_flux.py", line 153, in __init__
> >     self.pipe.transformer = transform(self.pipe.transformer)
> >   File "src/nexfort/compilers/transform_model.py", line 42, in nexfort.compilers.transform_model.transform_model
> >   File "src/nexfort/compilers/register_diffusers.py", line 15, in init nexfort.compilers.register_diffusers
> >   File "src/nexfort/nn/modules.py", line 14, in init nexfort.nn.modules
> >   File "src/nexfort/kernels/activations.py", line 43, in init nexfort.kernels.activations
> >   File "/home/shadeform/miniconda3/envs/image/lib/python3.10/site-packages/triton/runtime/jit.py", line 882, in jit
> >     return decorator(fn)
> >   File "/home/shadeform/miniconda3/envs/image/lib/python3.10/site-packages/triton/runtime/jit.py", line 871, in decorator
> >     return JITFunction(
> >   File "/home/shadeform/miniconda3/envs/image/lib/python3.10/site-packages/triton/runtime/jit.py", line 704, in __init__
> >     self.starting_line_number = inspect.getsourcelines(fn)[1]
> >   File "/home/shadeform/miniconda3/envs/image/lib/python3.10/inspect.py", line 1121, in getsourcelines
> >     lines, lnum = findsource(object)
> >   File "/home/shadeform/miniconda3/envs/image/lib/python3.10/inspect.py", line 940, in findsource
> >     file = getsourcefile(object)
> >   File "/home/shadeform/miniconda3/envs/image/lib/python3.10/inspect.py", line 817, in getsourcefile
> >     filename = getfile(object)
> >   File "/home/shadeform/miniconda3/envs/image/lib/python3.10/site-packages/torch/package/package_importer.py", line 712, in _patched_getfile
> >     return _orig_getfile(object)
> >   File "/home/shadeform/miniconda3/envs/image/lib/python3.10/inspect.py", line 797, in getfile
> >     raise TypeError('module, class, method, function, traceback, frame, or '
> > TypeError: module, class, method, function, traceback, frame, or code object was expected, got cython_function_or_method
> > ```
> 
> please install torch==2.4.0 and try Again

Name: torch
Version: 2.4.0

same issue happend tho, could you share us your environment list in  correct requirements.txt?

**Pull request merged**